src/store/actions.js

```diff
@@ -17,16 +19,13 @@ export const actions = {
     // but it finally works
   },
   async submit({ state, dispatch, commit }, fields) {
-    // parse the data for the different types of stocks
-    let submitData = InputParser.submitData(fields, state.activeTab);
+    console.log(state.activeTab, fields);
+    let stock = stockFactory(state.activeTab, fields);
 
-    let result = await api.create(submitData);
+    let result = await api.create(stock);
```

Comment:
```suggestion
    const result = await api.create(stock);
```
If you are going to change a variable directly later in a code block, use `let`. Else, use `const`.

In Javascript, as far as I know, it let you mutate an const declared variable/object ***properties***, but not the variable/object itself.

If you want to make both the variable/object and the properties immutable, see - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze